python/cudf/cudf/_lib/types.pxd

```diff
@@ -2,10 +2,15 @@
 
 from libc.stdint cimport int32_t
 from libcpp cimport bool
+from cudf._lib.cpp.column.column_view cimport column_view
+from cudf._lib.cpp.lists.lists_column_view cimport lists_column_view
+
 
 ctypedef bool underlying_type_t_order
 ctypedef bool underlying_type_t_null_order
 ctypedef bool underlying_type_t_sorted
 ctypedef int32_t underlying_type_t_interpolation
 ctypedef int32_t underlying_type_t_type_id
 ctypedef bool underlying_type_t_null_policy
+
+cdef dtype_from_column_view(column_view cv)
```

Comment:
`except *`?

Reply:
Since this function returns a Python type, it can propagate exceptions without `except *`. Only cdef functions that have a C return value need to marked `except *`.

Reply:
Do we need `except +` then to catch C++ exceptions?

Reply:
Only `extern` functions can throw C++ exceptions, and I guess the assumption is that any extern functions being called within `dtype_from_column_view` have been marked `except +` if they can throw. If that's the case, we don't need an `except +` here. I think a good convention to follow is:

* Mark all extern C++ functions that can throw `except +`
* Mark all `cdef` (non-extern) functions that return a C value `except *`